### ISSUE TITLE
Better clarity on inline targets for Target Size

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag#2767]
This is an attempt to split off the conversation around text links from https://github.com/w3c/wcag/issues/2695. It became clear during that lengthy discussion that, regardless of how to calculate the target offset, there was general agreement that we need better clarity/consistency on which text links on a page meet either the inline or user agent exception. 
NOTE: This issue is intentionally isolated to the inline exception. It is my hope that a separate issue will be opened on the user agent exception, in regard to links.

The reality:
1. Essentially, while most text links are long enough to be 24px wide, most are not going to be 24px high (and so will need to meet an exception or fail).
2. Text links that are not finely positioned by authors are likely to reposition within the content (through user agent text reflow) such that it is difficult for an author to anticipate all the ways they may end up directly above or below another text link on a page.
3. Typical text size plus line height is insufficient to meet 24px high (i.e., 12pt type with [typical line height of 120-145%](https://practicaltypography.com/line-spacing.html#) is only going to get to the low 20s)
4. Short text links exist, including superscript numerals used to references footnotes, that will not be 24px wide.

#### Inline exception
To avoid requiring _all_ text links to meet the 24x24 size, [an inline exception](https://w3c.github.io/wcag/understanding/target-size-minimum.html) was created. It reads:
> Inline: The target is in a sentence or block of text;

While there is no link to a definition of sentence or block of text, there _is_ a definition of [blocks of text](https://w3c.github.io/wcag/guidelines/22/#dfn-blocks-of-text) in WCAG:
> more than one sentence of text

So, the close reading of current wording is that a link is only excepted if it is in a sentence ("block of text" really only additionally covering a text link that spans more than one sentence).

Sentence is not defined. This [definition from Oxford](https://www.google.com/search?q=definition+of+sentence&rlz=1C5GCEM_en___US996&oq=de&aqs=chrome.0.69i59j0i20i131i263i433i512j69i57j46i39i175i199j69i60l4.2425j0j7&sourceid=chrome&ie=UTF-8) covers what I think is most people's interpretation of a sentence:
> a set of words that is complete in itself, typically containing a subject and predicate, conveying a statement, question, exclamation, or command, and consisting of a main clause and sometimes one or more subordinate clauses.

[Miriam-Webster](https://www.merriam-webster.com/dictionary/sentence) is looser on a 'complete' set of words, but does add in the notion of terminal punctuation:
> a word, clause, or phrase or a group of clauses or phrases forming a syntactic unit which expresses an assertion, a question, a command, a wish, an exclamation, or the performance of an action, that in writing usually begins with a capital letter and concludes with appropriate end punctuation, and that in speaking is distinguished by characteristic patterns of stress, pitch, and pauses

From these two, I believe we can derive the following from the current "inline" exception, at least in English:
- where a piece of text does not have terminal punctuation (period, exclamation mark, question mark) it may not be a sentence
- where a piece of text is a sentence fragment, it is likely not a sentence

Reviewing the [W3c wikipedia page](https://en.wikipedia.org/wiki/World_Wide_Web_Consortium) cited in the other issue, I believe the following links can be said to _not_ meet the inline exception:
1. All left-side navigation
2. All links that are essentially top nav ("talk", "Contributions", "Edit", "View history")
3. Links in square bracket essentially fulfilling a button/toolbar role (e.g., "edit" at start of each section)
4. All inks in the Contents section
5. Coordinates
6. All the links in the W3C table/sidebar on the right side (e.g., "Standards organization")
7. See also references ("See also: History of the World Wide Web")
8. all numbered footnote references that land _after_ the terminal punctuation in a sentence. This is ALL of them until one in the Administration section that occurs inline.
9. All the links in the Standards section, including all the links in the bullets, none of which end in terminal punctuation, and all of which are sentence fragments.
10. All the links in the External links section, including those listed in the Categories

The following links I had questions about:
- Footnotes, even though they are outside the sentence punctuation, _can_ be between 2 sentences in the same paragraph, which makes it pretty hard to say that they aren't in a block of text. 
- The "^" links that precede the text in the References section. Are they part of the sentence? I'm inclined to say "no", since they do not semantically have anything to do with the sentence syntax. They _do_ follow the reference number and are part of a string of text that ends in a period, though.
- Sentence fragments that are targets in the Reference sections which nonetheless end in periods (e.g., "W3C Contact", "List of Offices")
- The lowercase superscript letters that precede references 2, 3, and 8
> ^ a b c "Current Members - W3C" . World Wide Web Consortium. Retrieved 21 March 2022.

#### Overall, consistent results

In my opinion, this 'tight' reading of the inline definition seems to give pretty good results. 
- Links in paragraphs get a pass. These are going to be the ones most likely to flow in a way difficult for an author to control.
- Links in navigation sections don't; for the most part, these _are_ positioned by the author, so easier to ensure enough spacing.

I believe the most problematic areas are links in bulleted lists and footnotes.

Bulleted list items are very frequently sentence fragments. On the web lists are wildly inconsistent in regard to terminal punctuation, even within the same set of list items. It seems very arbitrary that for two link targets identical in size, one will pass because it has a non-interactive period behind it, where the other one fails. Countering this is the fact that a lot of text links used in a different role (button, tablist, navigation, tree) will be put in a list.

Footnotes deserve their own issue investigation. The strongest argument for _not_ excluding them is that they are probably the most difficult to operate link targets on the web. Countering this is centuries of convention. They are intentionally small so as not to disrupt the reading of content, which they merely supplement. Making them larger could have significant impact on comprehension, especially in regard to some users with cognitive and learning disabilities.

**Comment by mbgower** [comment]
For those who can see, I exported the wikipedia page to PDF and roughly highlighted the links which I don't think would be excepted, base on the above 'tight' reading of the inline exception language
[World Wide Web Consortium - Wikipedia.pdf](https://github.com/w3c/wcag/files/9944226/World.Wide.Web.Consortium.-.Wikipedia.pdf)

**Comment by bruce-usab** [comment]
Thanks for breaking this off from https://github.com/w3c/wcag/issues/2695 !

I would note that presently 2.2 2.5.8 is using these same two exceptions as as 2.1 2.5.5.  I think AGWG already made the call.
: https://www.w3.org/WAI/WCAG21/Understanding/target-size.html

--- 
Inline: Content displayed can often be reflowed based on the screen width available. This is known as responsive design and makes it easier to read since you do not need to scroll both horizontally and vertically. In reflowed content, the targets can appear anywhere on a line and can change position based on the width of the available screen. Since targets can appear anywhere on the line, the size cannot be larger than the available text and spacing between the sentences or paragraphs, otherwise the targets could overlap. It is for this reason targets which are contained within one or more sentences are excluded from the target size requirements.

Note:  A footnote or an icon within or at the end of a sentence is considered to be part of a sentence and therefore are excluded from the minimum target size.
--- 

So I think that does not exempt link text in, for example, the Wikipedia [categories](https://en.wikipedia.org/wiki/World_Wide_Web_Consortium#catlinks) block.

> User Agent Control: If the size of the target is not modified by the author through CSS or other size properties, then the target does not need to meet the target size of 44 by 44 CSS pixels.

Welp, that one does not seem so great to me!  Even if the Wikipedia category link text didn't specific size, the body text size is probably set.  Does this exception apply if an author uses _tiny_ jpegs for buttons -- so long as the author lets the browser determine the rendering?

**Comment by alastc** [comment]
Ok, so it sounds like the inline exception is doing what is needed, in general.

> I believe the most problematic areas are links in bulleted lists and footnotes.

Bruce noted the footnotes aspect was already mentioned (decided) in the AAA version.

Regarding the list aspect, e.g.
<img src="https://user-images.githubusercontent.com/216630/201693175-6f0147ea-d544-405e-bba5-5c2e54deec05.png" alt="screenshot from wikipedia of a list of links, some have non-link text, some don't.">

Many of those appear to be part of a sentence?

Perhaps we can add some clarification so that:

- "content" lists are considered "inline", i.e. in the main content of the page, not styled to be navigation etc.
- Navigation / buttons which happen to use list markup are not covered.

Regarding the user-agent exception, this was intended to cover things like select-drop-downs which can't be re-sized, or default checkboxes. I don't think that helps (excepts) text links which are very likely to be affected by any text-sizing CSS.

**Comment by alastc** [comment]
We have a [working doc](https://docs.google.com/document/d/1BFNDFnbU9CizswP4oEICpFHeqt7qXg9kRt2QEDoUXSM/edit#) outlining various options. 

The favoured option from a small meeting was:
**Inline**: The target is a text link dependent on the line-height of non-link text.
 
There are a couple of scenarios that could be interpreted in either direction (e.g. bullets and numbers in lists), that we’d need to outline in the understanding doc. More details in the doc.
